buildrunner/__init__.py

```diff
@@ -694,6 +694,18 @@ def run(self):  # pylint: disable=too-many-statements,too-many-branches,too-many
             self._write_artifact_manifest()
 
             _docker_client = docker.new_client(timeout=self.docker_timeout)
+
+            # cleanup the source image
+            if self._source_image:
```

Comment:
This order of image deletions avoids the situation of failing to remove a image because there is a dependent child image.